packages/manager/src/hooks/useToastNotifications.tsx

```diff
@@ -17,10 +18,28 @@ const formatLink = (text: string, link: string, handleClick?: () => void) => {
   );
 };
 
+type PersistType = 'both' | 'failure' | 'success';
+
 interface Toast {
+  /**
+   * If defined, this message will be displayed when the event fails.
+   */
   failure?: ((event: Event) => string | undefined) | string;
+  /**
+   * If true, the toast will be displayed with an error variant.
+   */
+  invertVariant?: boolean;
+  /**
+   * If defined, this link will be displayed in the toast.
+   */
   link?: JSX.Element;
-  persistFailureMessage?: boolean;
+  /**
+   * If defined, the toast will persist until dismissed.
+   */
+  persist?: PersistType;
+  /**
+   * If defined, this message will be displayed when the event succeeds.
+   */
   success?: ((event: Event) => string | undefined) | string;
```

Comment:
Instead of having a top level `persist`, could we do something like this:

```suggestion
  success?: {
    message: ((event: Event) => string | undefined) | string;
    persist: boolean;
  }
```